krux_marathon_api/marathonapi.py

```diff
@@ -66,42 +64,41 @@ def read_config_file(self, config_file):
-        self.logger.info("Assigning json variables to app data")
-        self.logger.debug("Config file data: {}".format(config_file_data))
-        marathon_app_result.cpus                     = config_file_data["cpus"]
-        marathon_app_result.mem                      = config_file_data["mem"]
-        marathon_app_result.instances                = config_file_data["instances"]
-        marathon_app_result.disk                     = config_file_data["disk"]
-        marathon_app_result.cmd                      = config_file_data["cmd"]
-        marathon_app_result.backoff_factor           = config_file_data["backoff_factor"]
-        marathon_app_result.constraints              = config_file_data["constraints"]
-        marathon_app_result.container                = config_file_data["container"]
-        marathon_app_result.dependencies             = config_file_data["dependencies"]
-        marathon_app_result.env                      = config_file_data["env"]
-        marathon_app_result.executor                 = config_file_data["executor"]
-        marathon_app_result.health_checks            = config_file_data["health_checks"]
-        marathon_app_result.labels                   = config_file_data["labels"]
-        marathon_app_result.max_launch_delay_seconds = config_file_data["max_launch_delay_seconds"]
-        ### If you define ports then blank the port definitions, but if you define
-        ### port_definitions they override ports, since port_definitions are more
-        ### specific (i.e. protocol and etc)
-        if config_file_data.get("ports"):
-            marathon_app_result.ports                = config_file_data["ports"]
-            marathon_app_result.port_definitions     = None
-        if config_file_data.get("port_definitions"):
-            marathon_app_result.port_definitions     = config_file_data["port_definitions"]
-            marathon_app_result.ports                = None
-        marathon_app_result.require_ports            = config_file_data["require_ports"]
-        marathon_app_result.store_urls               = config_file_data["store_urls"]
-        marathon_app_result.upgrade_strategy         = config_file_data["upgrade_strategy"]
-        marathon_app_result.uris                     = config_file_data["uris"]
-        marathon_app_result.user                     = config_file_data["user"]                 #updates confirmed
-        self.logger.debug("Marathon app object: {}".format(marathon_app_result))
+        ### value for if there is a change in our json file from server values
+        changes_in_json = False
+
+        ### iterate through our json file's values and compare them to the values
+        ### on the marathon server
+        for attribute, value in config_file_data.iteritems():
+            ### upgrade_strategy value returns a class an not a json, extra work
+            ### needs to be done to convert this to a comparable value
+            if attribute == 'upgrade_strategy':
```

Comment:
I believe other attributes are returned as MarathonResource sub classes. How about instead of possibly chasing attribute by attribute, you do something like:

```python
if isinstance(value, marathon.MarathonObject):
  marathon_app_result_original = getattr(marathon_app_result, attribute).to_json()
  value_json = json.dumps(value)
```

etc etc, makes sense?